### PR TITLE
TiKV config: add docs for raft engine configurations

### PR DESCRIPTION
**Pull request by TomShawn**
### What is changed, added or deleted? (Required)
TiKV config: add docs for raft engine configurations

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [x] master (the latest development version)
- [x] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)
- [ ] v4.0 (TiDB 4.0 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from: https://github.com/pingcap/docs/pull/7354 https://github.com/pingcap/docs/pull/7431
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: tikv-configuration-file.md [hunk 1]
```diff
@@ -1294,6 +1294,61 @@ raftdb 相关配置项。
...
++ 默认值：`""`
+
+### `batch-compression-threshold`
+
++ 指定日志批处理的阈值大小。大于此配置的日志批次将被压缩。如果将此配置项设置为 `0`，则禁用压缩。
++ 默认值：`"8KB"`
+
+### `bytes-per-sync`
+
++ 指定缓冲写入的最大累积大小。当超过此配置值时，缓冲的写入将被刷写到磁盘。
```
Comment (tabokie) [inline tikv-configuration-file.md:1320]:
```suggestion
+ 指定缓存写入的最大累积大小。当超过此配置值时，缓存的写入将被刷写到磁盘。
```

File: tikv-configuration-file.md [hunk 2]
```diff
@@ -1294,6 +1294,61 @@ raftdb 相关配置项。
...
++ 如果将此配置项设置为 `0`，则禁用增量同步。
++ 默认值：`"4MB"`
+
+### `target-file-size`
+
++ 指定日志文件的最大大小。当日志文件大于此值时，将对其进行轮换。
```
Comment (tabokie) [inline tikv-configuration-file.md:1326]:
```suggestion
+ 指定日志文件的最大大小。当日志文件大于此值时，将对其进行轮转。
```

File: tikv-configuration-file.md [hunk 3]
```diff
@@ -1294,6 +1294,61 @@ raftdb 相关配置项。
...
++ 默认值：`"128MB"`
+
+### `purge-threshold`
+
++ 指定主日志队列的阈值大小。当超过此配置值时，主日志队列将被清除。
```
Comment (tabokie) [inline tikv-configuration-file.md:1331]:
```suggestion
+ 指定主日志队列的阈值大小。当超过此配置值时，将对主日志队列执行垃圾回收。
```

File: tikv-configuration-file.md [hunk 4]
```diff
@@ -1294,6 +1294,61 @@ raftdb 相关配置项。
...
++ 默认值：`"10GB"`
+
+### `recovery-mode`
+
++ 确定在恢复过程中如何处理文件损坏。
```
Comment (tabokie) [inline tikv-configuration-file.md:1336]:
```suggestion
+ 确定在日志恢复过程中如何处理文件损坏。
```

**Review by tabokie — APPROVED** [review]

File: tikv-configuration-file.md [hunk 5]
```diff
@@ -1294,6 +1294,62 @@ raftdb 相关配置项。
 + WAL 存储目录。
 + 默认值：/tmp/tikv/store
 
+## raft-engine
+
+Raft engine 相关的配置项。
```
Comment (hfxsd) [inline tikv-configuration-file.md:1299]:
```suggestion
Raft Engine 相关的配置项。
```

File: tikv-configuration-file.md [hunk 6]
```diff
@@ -1294,6 +1294,62 @@ raftdb 相关配置项。
...
+### `enable`
+
++ 决定是否使用 Raft Engine 来存储 Raft 日志。开启该配置项后，`raftdb` 的配置不再生效。
++ 默认值：`"false"`
+
+### `dir`
+
++ 存储 Raft 日志文件的目录。如果该目录不存在，则在启动 TiKV 时创建该目录。
++ 如果未设置此配置，则使用 `{data-dir}/raft-engine`。
++ 如果您的机器上有多个磁盘，建议将 Raft Engine 的数据存储在不同的磁盘上，以提高 TiKV 性能。
```
Comment (hfxsd) [inline tikv-configuration-file.md:1310]:
```suggestion
+ 如果你的机器上有多个磁盘，建议将 Raft Engine 的数据存储在不同的磁盘上，以提高 TiKV 性能。
```
原文是 a different disk，似乎是说 “数据和 Raft Engine 不要在同一个磁盘上”。中文可能会误解为把数据分散存储到多个磁盘。建议中文这句话调整下。

**Review by hfxsd — APPROVED** [review]
Rest LGTM

File: tikv-configuration-file.md [hunk 7]
```diff
@@ -1294,6 +1294,62 @@ raftdb 相关配置项。
...
+### `enable`
+
++ 决定是否使用 Raft Engine 来存储 Raft 日志。开启该配置项后，`raftdb` 的配置不再生效。
++ 默认值：`"false"`
+
+### `dir`
+
++ 存储 Raft 日志文件的目录。如果该目录不存在，则在启动 TiKV 时创建该目录。
++ 如果未设置此配置，则使用 `{data-dir}/raft-engine`。
++ 如果你的机器上有多个磁盘，建议将 Raft Engine 的数据存储在不同的磁盘上，以提高 TiKV 性能。
```
Comment (TomShawn) [inline tikv-configuration-file.md:1310]:
```suggestion
+ 如果你的机器上有多个磁盘，建议将 Raft Engine 的数据存储在单独的磁盘上，以提高 TiKV 性能。
```

**Pull request merged**